CRM/Utils/Hook.php

```diff
@@ -3081,41 +3081,6 @@ public static function alterCustomFieldDisplayValue(&$displayValue, $value, $ent
     );
   }
 
-  /**
-   * Alter APIv4 route permissions based on the Entity and Action
-   *
-   * This is an experimental hook intended to *relax* the requirement
-   * for "access AJAX API" when calling public-oriented APIs.
-   *
-   * Historically, when APIv2/v3 were first exposed to an HTTP interface, using
-   * the HTTP interface required an extra permission "access AJAX API". This is a
-   * broad hedge against security flaws within those API's. In the current APIv4
-   * era, security concerns are often baked into each API, so there is a debate about
-   * whether "access AJAX API" serves a purpose or just makes
-   * administration/development more complicated. (So far, there's more support
-   * for the latter.)
-   *
-   * This hook might foreseeably be abandoned either...
-   *
-   * - if it is found that "access AJAX API" guard is not needed for APIv4.
-   * - if the policy is moved into metadata.
-   *
-   * @param array|string $permissions
-   * @param string $entity
-   * @param string $action
-   *
-   * @return mixed
-   */
-  public static function alterApiRoutePermissions(&$permissions, $entity, $action) {
```

Comment:
I did a search of `universe` and only found one other consumer of this hook (cc @MegaphoneJon):

https://github.com/MegaphoneJon/com.megaphonetech.monitoring/blob/86e0776aafb46e3181049551f8bb700f1aa0cf54/monitoring.php#L78

However, I think this is OK -- the route-level guard goes away, but the API-level guard is still there. And `hook_alterAPIPermissions` provides a similar adjustment for the API-level guard.